centaurCwlRunner/src/main/scala/centaur/cwl/PAPIPreprocessor.scala

```diff
@@ -0,0 +1,121 @@
+package centaur.cwl
+import io.circe.optics.JsonPath
+import io.circe.optics.JsonPath._
+import io.circe.{Json, JsonObject, yaml}
+
+/**
+  * Tools to pre-process the CWL workflows and inputs before feeding them to Cromwell so they can be executed on PAPI.
+  */
+object PAPIPreprocessor {
+  // GCS directory where inputs for conformance tests are stored
+  private val gcsPrefix = "gs://centaur-cwl-conformance/cwl-inputs/"
+
+  // Default docker pull image
+  val DefaultDockerPull = "dockerPull" -> Json.fromString("ubuntu:latest")
+  
+  // Default docker image to be injected in a pre-existing requirements array
+  private val DefaultDockerRequirement: Json = {
+    Json.obj(
+      "class" -> Json.fromString("DockerRequirement"),
+      DefaultDockerPull
+    )
+  }
+
+  // Requirements array with default docker requirement
+  private val DefaultDockerRequirementList: Json = {
+    Json.obj(
+      "requirements" -> Json.arr(DefaultDockerRequirement)
+    )
+  }
+
+  // Parse value, apply f to it, and print it back to String using the printer
+  private def process(value: String, f: Json => Json, printer: Json => String) = {
+    yaml.parser.parse(value) match {
+      case Left(error) => throw new Exception(error.getMessage)
+      case Right(json) => printer(f(json))
+    }
+  }
+
+  // Process and print back as YAML
+  private def processYaml(value: String)(f: Json => Json) = process(value, f, yaml.Printer.spaces2.pretty)
+
+  // Process and print back as JSON
+  private def processJson(value: String)(f: Json => Json) = process(value, f, io.circe.Printer.spaces2.pretty)
+
+  // Prefix the string at "key" with the gcs prefix
+  private def prefixLocationWithGcs(key: String): Json => Json = root.selectDynamic(key).string.modify(gcsPrefix + _)
+
+  // Prefix "location" and "path"
+  private val prefix = prefixLocationWithGcs("location").compose(prefixLocationWithGcs("path"))
```

Comment:
We shouldn't ever need to change `path` to be GCS. That's for the localized path on the VM (vs the universal resource **Location** being a fully qualified, accessible path)

Reply:
The above comment is already an open ticket if someone wants to pick it up: #3132